packages/agreement-process/src/services/agreementService.ts

```diff
@@ -711,26 +703,33 @@ export async function cloneAgreementLogic({
 
   validateCertifiedAttributes(descriptor, consumer.data);
 
-  const newAgreement = await createAgreementLogic(
-    {
-      eserviceId: agreementToBeCloned.data.eserviceId,
-      descriptorId: agreementToBeCloned.data.descriptorId,
-    },
-    authData,
-    agreementQuery,
-    eserviceQuery,
-    tenantQuery
-  );
+  const newAgreement: Agreement = {
+    id: uuidv4(),
+    eserviceId: agreementToBeCloned.data.eserviceId,
+    descriptorId: agreementToBeCloned.data.descriptorId,
+    producerId: agreementToBeCloned.data.producerId,
+    consumerId: agreementToBeCloned.data.consumerId,
+    consumerNotes: agreementToBeCloned.data.consumerNotes,
+    verifiedAttributes: [],
+    certifiedAttributes: [],
+    declaredAttributes: [],
+    state: agreementState.draft,
+    createdAt: new Date(),
+    consumerDocuments: [],
+    stamps: {},
+  };
+
+  const createEvent = toCreateEventAgreementAdded(newAgreement);
 
   const docEvents = await createAndCopyDocumentsForClonedAgreement(
-    newAgreement.streamId,
+    createEvent.streamId,
     agreementToBeCloned.data,
     0,
     fileCopy
   );
 
   return {
-    streamId: newAgreement.streamId,
-    events: [newAgreement, ...docEvents],
+    streamId: createEvent.streamId,
+    events: [createEvent, ...docEvents],
```

Comment:
We have double-checked that in this case, the stream ID to use is the one from the cloned agreement and not the one from the new agreement.

Reply:
never mind, I didn't notice the `createEvent` assign